packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts

```diff
@@ -62,28 +82,26 @@ describe('HttpCachingProxy', () => {
     await givenRunningProxy({logError: false});
     await expect(
       makeRequest({
-        uri: 'http://does-not-exist.example.com',
-        proxy: proxy.url,
-        resolveWithFullResponse: true,
+        url: 'http://does-not-exist.example.com',
+        proxy: getProxyConfig(),
       }),
     ).to.be.rejectedWith(
       // The error can be
       // '502 - "Error: getaddrinfo EAI_AGAIN does-not-exist.example.com:80"'
       // '502 - "Error: getaddrinfo ENOTFOUND does-not-exist.example.com'
-      /502 - "Error\: getaddrinfo/,
+      /502 - getaddrinfo/,
     );
   });
 
   it('reports timeout error for HTTP requests', async function() {
     await givenRunningProxy({logError: false, timeout: 1});
     await expect(
       makeRequest({
-        uri:
+        url:
           'http://www.mocky.io/v2/5dade5e72d0000a542e4bd9c?mocky-delay=1000ms',
-        proxy: proxy.url,
-        resolveWithFullResponse: true,
```

Comment:
`resolveWithFullResponse` is request's counter-part of Axios' `validateStatus`. Have you considered configuring `validateStatus` on a per-test basis, as we are doing with `resolveWithFullResponse` now?